### DISCUSSION TITLE
SPAN_LIMIT_PER_SEGMENT doesn't work?

### DISCUSSION THREAD

**Discussion by liyanyan-dev** [apache/skywalking#8064] [Q&A]
version: skywalking-8.1.0
os: centos7
jdk: jdk1.8

Why push  NoopSpan into the stack when the SPAN_LIMIT_PER_SEGMENT is exceeded in TracingContext.java?  The limit doesn't actually work ?
![image](https://user-images.githubusercontent.com/54348450/140044271-81b67ccd-40d7-4a54-b6b9-d6abe6b8dca7.png)

**Comment by wu-sheng** [comment]
Just notice, open source community expects you to read and understand. We can't afford to explain code details. Sorry.

If others want to discuss code details, feel free to go
